std/algorithm/comparison.d

```diff
@@ -699,6 +699,9 @@ int cmp(alias pred = "a < b", R1, R2)(R1 r1, R2 r2) if (isSomeString!R1 && isSom
     assert(result > 0);
 }
 
+enum areEquable(T, U) = is(typeof({ return T.init == U.init; })); // TODO move to std.traits
+enum haveEquableLengths(T, U) = is(typeof({ return T.init.length == U.init.length; })); // TODO move to std.traits
```

Comment:
Just assume lengths are always size_t hence no need for hasLengthComparable etc.